usecases/config/config_handler.go

```diff
@@ -58,6 +58,9 @@ const (
 	DefaultBM25b  = float32(0.75)
 )
 
+// DefaultQueryMaxRefDepth describes the max depth of nested references in a query
+const DefaultQueryMaxRefDepth = 3
```

Comment:
hmm, shouldn't this setting be defined like [QueryMaximumResults](https://github.com/weaviate/weaviate/blob/main/usecases/config/config_handler.go#L96) and be a part of Config struct?

Reply:
in traverser you have access to this Config struct like so: `t.config.Config.QueryMaximumResults`

Reply:
I agree that would be preferred, and that was also my first thought. But the db config isn't available until much later down the line:
`Traverser` -> `Explorer` -> `objectsSearcher`. 

And then `objectsSearcher` has like 8 different entrypoints into the `DB`:

```go
type objectsSearcher interface {
	hybridSearcher

	// GraphQL Get{} queries
	Search(ctx context.Context, params dto.GetParams) ([]search.Result, error)
	VectorSearch(ctx context.Context, params dto.GetParams) ([]search.Result, error)

	// GraphQL Explore{} queries
	CrossClassVectorSearch(ctx context.Context, vector []float32, targetVector string, offset, limit int,
		filters *filters.LocalFilter) ([]search.Result, error)

	// Near-params searcher
	Object(ctx context.Context, className string, id strfmt.UUID,
		props search.SelectProperties, additional additional.Properties,
		properties *additional.ReplicationProperties, tenant string) (*search.Result, error)
	ObjectsByID(ctx context.Context, id strfmt.UUID, props search.SelectProperties, additional additional.Properties, tenant string) (search.Results, error)
}

type hybridSearcher interface {
	SparseObjectSearch(ctx context.Context, params dto.GetParams) ([]*storobj.Object, []float32, error)
	DenseObjectSearch(context.Context, string, []float32, string, int, int,
		*filters.LocalFilter, additional.Properties, string) ([]*storobj.Object, []float32, error)
	ResolveReferences(ctx context.Context, objs search.Results, props search.SelectProperties,
		groupBy *searchparams.GroupBy, additional additional.Properties, tenant string) (search.Results, error)
}
```

I was hoping to handle this case as early on as possible, in a simple way that won't miss any edge cases

Reply:
Great point, will make the change